circuit/src/dynamic_sets/mod.rs

```diff
@@ -65,27 +65,23 @@ impl<const NUM_NEIGHBOURS: usize, const NUM_ITER: usize, const INITIAL_SCORE: u1
 {
 	/// Constructs a new EigenTrustSet circuit
 	pub fn new(
-		pks: Vec<PublicKey>, signatures: Vec<Signature>, op_pks: Vec<Vec<PublicKey>>,
-		ops: Vec<Vec<Scalar>>,
+		pks: Vec<UnassignedPublicKey>, signatures: Vec<UnassignedSignature>,
+		op_pks: Vec<Vec<UnassignedPublicKey>>, ops: Vec<Vec<Scalar>>,
```

Comment:
I think it will better DX to pass normal values here, so: `Vec<PublicKey>` and `Vec<Signature>`.
Then inside this function you create the unassigned values from them.
You already created `UnassignedPublicKey::from(x)` and `UnassignedSignature::from(x)`, so this should be easy